numpy/lib/tests/test_shape_base.py

```diff
@@ -662,6 +662,13 @@ def test_kron_smoke(self):
 
         assert np.array_equal(np.kron(a, b), k), "Smoke test for kron failed"
 
+    def test_kron_ma(self):
+        x = np.ma.array([[1, 2], [3, 4]], mask=[[0, 1], [1, 0]])
+        k = np.ma.array(np.diag([1, 4, 4, 16]),
+                mask=~np.array(np.identity(4), dtype=bool))
+
+        assert np.ma.allequal(k, np.kron(x, x)), "Kron for ma failed"
```

Comment:
`assert_array_equal` explicitly tests that something is the right subclass as well.

Reply:
ohh ok, will do :+1:

Reply:
Thanks for the info!